File: client/bcephal-client-model/Billing/Invoices/InvoiceBrowserData.cs
using Bcephal.Models.Base;
using Newtonsoft.Json;
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;
using System.Threading.Tasks;

namespace Bcephal.Models.Billing.Invoices
{
    public class InvoiceBrowserData : BrowserData
    {
        public string Reference { get; set; }

        public string RunNumber { get; set; }

        public string ClientNumber { get; set; }

        public string ClientName { get; set; }

        public DateTime? InvoiceDate { get; set; }

        public DateTime? BillingDate { get; set; }

        public decimal? AmountWithoutVat { get; set; }

        public decimal? VatAmount { get; set; }

        public decimal? TotalAmount { get; set; }

        public int Version { get; set; }

        public bool ManuallyModified { get; set; }

        public string Status { get; set; }

        [JsonIgnore]
        public InvoiceStatus InvoiceStatus
        {
            get { return !string.IsNullOrEmpty(Status) ? InvoiceStatus.GetByCode(Status) : null; }
            set { this.Status = value != null ? value.code : null; }
        }

        public string MailStatus { get; set; }

        [JsonIgnore]
        public MailSendingStatus MailSendingStatus
        {
            get { return !string.IsNullOrEmpty(MailStatus) ? MailSendingStatus.GetByCode(MailStatus) : null; }
            set { this.MailStatus = value != null ? value.code : null; }
        }

    }
}

File: server/bcephal-billing/src/main/java/com/moriset/bcephal/billing/domain/InvoiceBrowserData.java
/**
 * 
 */
package com.moriset.bcephal.billing.domain;


import java.math.BigDecimal;
import java.util.Date;

import com.fasterxml.jackson.databind.annotation.JsonDeserialize;
import com.fasterxml.jackson.databind.annotation.JsonSerialize;
import com.moriset.bcephal.domain.BrowserData;
import com.moriset.bcephal.domain.MailSendingStatus;
import com.moriset.bcephal.utils.JsonDateDeserializer;
import com.moriset.bcephal.utils.JsonDateSerializer;

import lombok.Data;
import lombok.EqualsAndHashCode;

/**
 * @author Joseph Wambo
 *
 */
@Data
@EqualsAndHashCode(callSuper=false)
public class InvoiceBrowserData extends BrowserData {

	@JsonSerialize(using = JsonDateSerializer.class)
	@JsonDeserialize(using = JsonDateDeserializer.class)
	private Date billingDate;
	
	@JsonSerialize(using = JsonDateSerializer.class)
	@JsonDeserialize(using = JsonDateDeserializer.class)
	private Date invoiceDate;
	
	private String reference;
	
	private String runNumber;
	
	private String clientNumber;
	
	private String clientName;
	
	private BigDecimal amountWithoutVat;
	
	private BigDecimal vatAmount;
	
	private int version;
	
	private boolean manuallyModified;
	
	private InvoiceStatus status;
	
	private MailSendingStatus mailStatus;
	
	
	public InvoiceBrowserData(Invoice invoice) {
		super(invoice);
		this.reference = invoice.getReference();
		this.runNumber = invoice.getRunNumber();
		this.billingDate = invoice.getBillingDate();
		this.invoiceDate = invoice.getInvoiceDate();
		this.clientNumber = invoice.getClientNumber();
		this.clientName = invoice.getClientName();
		this.amountWithoutVat = invoice.getAmountWithoutVat();
		
		this.vatAmount = invoice.getVatAmount();
		this.version = invoice.getVersion();
		this.manuallyModified = invoice.isManuallyModified();
		this.status = invoice.getStatus();
		this.mailStatus = invoice.getMailStatus();
	}
}
